docs/standard/io/buffers.md

```diff
@@ -0,0 +1,188 @@
+- Arithmetic can't be performed on `SequencePosition` without the `ReadOnlySequence<T>`. That means doing basic things like `position++` is written `ReadOnlySequence<T>.GetPosition(position, 1)`.
+- `GetPosition(long)` does **not** support negative indexes. That means it's impossible to get the second to last character without walking all segments.
+- Two `SequencePosition` can't be compared, making it difficult to:
+  - Know if one position is greater than or less than another position.
+  - Write some parsing algorithms.
+- `ReadOnlySequence<T>` is bigger than an object reference and should be passed by [in](/dotnet/csharp/language-reference/keywords/in-parameter-modifier) or [ref](/dotnet/csharp/language-reference/keywords/ref) where possible. Passing `ReadOnlySequence<T>` by `in` or `ref` reduces copies of the [struct](../../csharp/language-reference/keywords/struct.md).
+- Empty segments:
+  - Are valid within a `ReadOnlySequence<T>`
+  - Can appear when iterating using `ReadOnlySequence<T>.TryGet` 
+  - Can appear slicing the sequence using `ReadOnlySequence<T>.Slice()` with `SequencePosition`(s).
+
+## SequenceReader\<T\>
+
+<xref:System.Buffers.SequenceReader%601>:
+
+- Is a new type that was introduced in .NET Core 3.0 to simplify the processing of a `ReadOnlySequence<T>`.
+- Unifies the differences between a single segment `ReadOnlySequence<T>` and multi-segment `ReadOnlySequence<T>`.
+- Provides helpers for reading binary and text data (`byte` and `char`) that may or may not be split across segments.
+
+There are built-in methods for dealing with processing both binary and delimited data. The following section demonstrates what those same methods look like with the `SequenceReader<T>`:
+
+### Access data
+
+`SequenceReader<T>` has methods for enumerating data inside of the `ReadOnlySequence<T>` directly. The following code is an example of processing a `ReadOnlySequence<byte>` a `byte` at a time:
+
+[!code-csharp[](~/samples/snippets/csharp/buffers/MyClass.cs?name=snippet8)]
+
+The `CurrentSpan` exposes the current segment's `Span`, which is similar to what was done in the method manually.
+
+### Position
+
+The following code is an example implementation of `FindIndexOf` using the `SequenceReader<T>`:
+
+[!code-csharp[](~/samples/snippets/csharp/buffers/MyClass.cs?name=snippet9)]
+
+### Process binary data
+
+The following example parses a 4-byte big-endian integer length from the start of the `ReadOnlySequence<byte>`.
+
+[!code-csharp[](~/samples/snippets/csharp/buffers/MyClass.cs?name=snippet11)]
+
+### Process text data
+
+[!code-csharp[](~/samples/snippets/csharp/buffers/MyClass.cs?name=snippet10)]
+
+### SequenceReader\<T\> problem areas
+
+- `SequenceReader<T>` is a mutable struct, it should always be passed by [reference](../../csharp/language-reference/keywords/ref.md).
```

Comment:
```suggestion
- Because `SequenceReader<T>` is a mutable struct, it should always be passed by [reference](../../csharp/language-reference/keywords/ref.md).
```